docker-entrypoint.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+set -e
+
+if [ $(echo "$1" | cut -c1) = "-" ]; then
+  echo "$0: assuming arguments for bitcoind"
+
+  set -- bitcoind "$@"
+fi
+
+if [ $(echo "$1" | cut -c1) = "-" ] || [ "$1" = "bitcoind" ]; then
+  mkdir -p "$BITCOIN_DATA"
+  chmod 700 "$BITCOIN_DATA"
+  chown -R bitcoin "$BITCOIN_DATA"
+
+  echo "$0: setting data directory to $BITCOIN_DATA"
+
+  set -- "$@" -datadir="$BITCOIN_DATA"
+fi
+
+if [ "$1" = "bitcoind" ] || [ "$1" = "bitcoin-cli" ] || [ "$1" = "bitcoin-tx" ]; then
+  echo 
+  if [ "$1" = "bitcoind" ]; then
+      ROOT_BITCOIN="/root/.bitcoin"
+      ROOT_BITCOIN_CONF="${ROOT_BITCOIN}/bitcoin.conf"
+      RPC_PORT=""
+      mkdir ${ROOT_BITCOIN}
+
+      if [ $(echo "$@"|grep -c -- '-regtest') -ne 0 ]; then
+          echo "$0: Link cookie from regtest subfolder"
+          ln -s "$BITCOIN_DATA/regtest/.cookie" "${ROOT_BITCOIN}/"
+          RPC_PORT="18443"
+      elif [ $(echo "$@"|grep -c -- '-testnet') -ne 0 ]; then
+          echo "$0: Link cookie from testnet3 subfolder"
+          ln -s "$BITCOIN_DATA/testnet3/.cookie" "${ROOT_BITCOIN}/"
+          RPC_PORT="18332"
+      else
+          echo "$0: Link cookie from bitcoin folder"
```

Comment:
This is the essential thing that we want right?

Reply:
[suggest] Instead of greppy thing maybe just test for the existence of the cookie
```sh
if test -e "$BITCOIN_DATA/testnet3/.cookie"; then
   ...
fi
They should only be there if the flag was passed right?

Reply:
The cookie will not yet exist